lighthouse-client/src/lighthouse.rs

```diff
@@ -200,25 +201,38 @@ impl<S> Lighthouse<S>
     }
     
     /// Performs a STREAM request to the given path with the given payload.
+    /// Automatically sends a STOP once dropped.
     #[tracing::instrument(skip(self, payload))]
-    pub async fn stream<P, R>(&mut self, path: &[&str], payload: P) -> Result<impl Stream<Item = Result<ServerMessage<R>>>>
+    pub async fn stream<P, R>(&self, path: &[&str], payload: P) -> Result<impl Stream<Item = Result<ServerMessage<R>>>>
     where
         P: Serialize,
         R: for<'de> Deserialize<'de> {
         let request_id = self.send_request(&Verb::Stream, path, payload).await?;
         let stream = self.receive_streaming(request_id).await?;
-        // TODO: Send STOP once dropped
-        Ok(stream)
+        Ok(stream.guard({
+            // Stop the stream on drop
+            let this = (*self).clone();
+            let path: Vec<_> = path.into_iter().map(|s| s.to_string()).collect();
+            move || {
+                tokio::spawn(async move {
+                    // TODO: Find a more elegant way to pass the path, ideally without
+                    // converting back and forth between Vec<String>, Vec<&str> and &[&str]
+                    let path: Vec<_> = path.iter().map(|s| &**s).collect();
```

Comment:
Implementation note: The (rather messy) second conversion would potentially be redundant with #15.